rpcserver.go

```diff
@@ -66,17 +96,24 @@ func (s *server) AddPeer(
 		return makePlainResponse(false, fmt.Sprintf("Failed to generate peer key/ID with seed: %v", req.Seed)), nil
 	}
 
-	var replyMsg string
-	err = s.node.AddPeer(spanctx, mAddr)
-	success := (err == nil)
-	if success {
-		replyMsg = fmt.Sprintf("Added Peer %v:%v, pid=%v!", req.Ip, req.Port, targetPID)
-		// Tag the span with peer info
-		logger.SetTag(spanctx, "Added peer", fmt.Sprintf("%v:%v", req.Ip, req.Port))
-	} else {
-		replyMsg = fmt.Sprintf("Failed to add Peer %v:%v, pid=%v!", req.Ip, req.Port, targetPID)
+	peerid, targetAddr, err := parseAddr(mAddr)
+	if err != nil {
+		logger.FinishWithErr(spanctx, fmt.Errorf("Failed to parse peer address: %s, err: %v", mAddr, err))
+		log.Printf("Failed to parse peer address: %s, err: %v", mAddr, err)
+		return makePlainResponse(false, fmt.Sprintf("Failed to parse peer address: %s, err: %v", mAddr, err)), nil
 	}
-	return makePlainResponse(success, replyMsg), nil
+	s.node.Peerstore().AddAddr(peerid, targetAddr, pstore.PermanentAddrTTL)
```

Comment:
We add peers with `PermanentAddrTTL` here. However, will the added peers also add us with `PermanentAddrTTL`? or just a shorter TTL?

Reply:
A shorter TTL. Though I guess a shorter TTL makes more sense since we probably don't want others to bloat our peer store size at will?

Reply:
I think that might depend on what users expect when they call the RPC `AddPeer`? Personally, I expect TTL to be permanent in that case.